data/src/main/java/com/safetyheads/akademiaandroida/data/network/repository/UserRepositoryImpl.kt

```diff
@@ -114,6 +121,167 @@ class UserRepositoryImpl : UserRepository {
         awaitClose { listener.isCanceled }
     }
 
+    override suspend fun changeName(
+        firstName: String,
+        lastName: String,
+        userUUID: String
+    ): Flow<Result<Boolean>> = callbackFlow {
+        try {
+            val listener = collectionReference.collection("users").document(userUUID)
+                .update(
+                    mapOf(
+                        "firstName" to firstName,
+                        "lastName" to lastName
+                    )
+                ).addOnCompleteListener {
+                    trySend(Result.success(true))
+                }.addOnFailureListener { e ->
+                    trySend(Result.failure(e))
+                }
+            awaitClose { listener.isCanceled }
+        } catch (e: Exception) {
+            trySend(Result.failure(e))
+        }
+    }
+
+    override suspend fun changeJobPosition(
```

Comment:
Good suggestion, but how to handle receiving flow in even viewmodel.  Create variables or some method that will receive the response and match it to the appropriate query?  Because we must have some control over whether it has been changed correctly.  How do you think to solve it?

Reply:
Looks good after your change 🤔